src/stores/apikey.ts

```diff
@@ -0,0 +1,27 @@
+import { writable } from 'svelte/store';
+import { manageAPIKey } from '@/services/api';
+
+export default function () {
+
+	const apikey = writable({});
+	const loading = writable(false);
+	const error = writable(false);
+
+	async function controlAPIKey(action) {
+		loading.set(true);
+		error.set(false);
+
+		try {
+			const result = await manageAPIKey(action);
+			apikey.set(result.apikey);
+
+		} catch(e) {
+			error.set(e);
+		}
+		loading.set(false);
+	}
+
+	controlAPIKey('get');
+
+	return [apikey, loading, error, controlAPIKey];
```

Comment:
@blokhin I don't like this idea
look at my idea
```typescript
const apiStore = writable<Promise<string>>(...)

// create api key
apiStore.set(apiKey) or apiStore.set({action: 'set', value: apiKey})

// delete api key
apiStore.delete() or apiSore.set({action: 'delete'})
```

read api key and show errors
```svelte
{#await $apiStore}
... waiting api key
{:then apiKey}
... show api key
{:catch error}
... show error
{/await}
```

Reply:
@pyoner this is not really obvious for me :laughing: actually there is `asyncable` store from @PaulMaly which might be even more appropriate here, but I still haven't got a chance to wrap my head around it...

Reply:
@blokhin look at [this](https://svelte.dev/repl/7f87e103dcf84662a6ad6cbdee3c99b7?version=3.59.1)

Reply:
@pyoner OK, I see your approach and have taken the arrow function expressions from it. Its only drawback is that the `key` variable is imported from the store implicitly, which is bad. However let's not spend more time on refactoring this.